drivers/usb/udc/udc_dwc2.c

```diff
@@ -1725,6 +1737,9 @@ static int udc_dwc2_disable(const struct device *dev)
 		return -EIO;
 	}
 
+	config->irq_disable_func(dev);
+	sys_clear_bits((mem_addr_t)&base->gahbcfg, USB_DWC2_GAHBCFG_GLBINTRMASK);
+
```

Comment:
The other changes in udc_dwc2_ep_disable() look like f4e95ccd48e51c6f8ea351de1fd2e32955d4ce99 fixes. Perhaps this should be moved to its own commit.

Reply:
Yes, the changes here are really fixes to f4e95ccd48e51c6f8ea351de1fd2e32955d4ce99. I don't think this commit should be splitted.